resources/cmake/ConnextDdsAddExample.cmake

```diff
@@ -0,0 +1,276 @@
+
+    _connextdds_add_example(
+        NAME name
+        LANG language
+        [DISABLE_SUBSCRIBER]
+        [CODEGEN_ARGS ...]
+    )
+
+Add a new example to be built. This method will call Codegen, create the
+CMake targets and set the dependencies.
+
+``IDL`` (required):
+    Name of the IDL file (without extension).
+``LANG`` (required):
+    Example language. Valid values are ``C``, ``C++``, ``C++03`` and ``C++11``.
+``PREFIX``:
+    Prefix name for the targets.
+``DISABLE_SUBSCRIBER``:
+    Disable the subscriber build.
+``DEPENDENCIES``:
+    Other target dependencies.
+``CODEGEN_ARGS":
+    Extra arguments for Codegen.
+
+Output targets:
+``folder name><_publisher_<lang>``
+    Target for the publisher application.
+``<folder name>_subscriber_<lang>``
+    Target for the subscriber application.
+``<folder name>_qos_profile_<lang>``
+    Target to copy the USER_QOS_PROFILES.xml (if exists).
+#]]
+
+include_guard(DIRECTORY)
+
+# Find the RTI Connext DDS libraries
+if(NOT RTIConnextDDS_FOUND)
+    find_package(RTIConnextDDS
+        "5.3.0"
+        REQUIRED
+        COMPONENTS
+            core
+    )
+endif()
+
+include(CMakeParseArguments)
+include(ConnextDdsArgumentChecks)
+include(ConnextDdsCodegen)
+
+
+function(connextdds_add_example)
+    set(optional_args DISABLE_SUBSCRIBER REQUIRE_QOS)
+    set(single_value_args IDL LANG PREFIX)
+    set(multi_value_args CODEGEN_ARGS DEPENDENCIES)
+    cmake_parse_arguments(_CONNEXT
+        "${optional_args}"
+        "${single_value_args}"
+        "${multi_value_args}"
+        ${ARGN}
+    )
+    connextdds_check_required_arguments(
+        _CONNEXT_IDL
+        _CONNEXT_LANG
+    )
+
+
+    # First, we call Codegen to generate the header and source files from the
+    # IDL
+    connextdds_rtiddsgen_run(
+        IDL_FILE
+            "${CMAKE_CURRENT_SOURCE_DIR}/${_CONNEXT_IDL}.idl"
+        OUTPUT_DIRECTORY
+            "${CMAKE_CURRENT_BINARY_DIR}/src"
+        LANG ${_CONNEXT_LANG}
+        ${_CONNEXT_CODEGEN_ARGS}
+    )
+
+    # If the GENERATE_EXAMPLE option was provided, Codegen will generate the
+    # source code for the publisher and the subscriber applications
+    # If we don't provide that parameter, we will use the source code provided
+    # in the repository for the publisher and the subscriber
+    connextdds_sanitize_language(LANG ${_CONNEXT_LANG} VAR lang_var)
+    if(GENERATE_EXAMPLE IN_LIST _CONNEXT_CODEGEN_ARGS)
+        set(publisher_src ${${_CONNEXT_IDL}_${lang_var}_PUBLISHER_SOURCES})
+        set(subscriber_src ${${_CONNEXT_IDL}_${lang_var}_SUBSCRIBER_SOURCES})
+    else()
+
+        set(extension "c")
+        if("${_CONNEXT_LANG}" STREQUAL "C++" OR
+            "${_CONNEXT_LANG}" STREQUAL "C++03" OR
+            "${_CONNEXT_LANG}" STREQUAL "C++11")
+            set(extension "cxx")
+        endif()
+
+        set(publisher_src
+            "${CMAKE_CURRENT_SOURCE_DIR}/${_CONNEXT_IDL}_publisher.${extension}"
+            ${${_CONNEXT_IDL}_${lang_var}_SOURCES}
```

Comment:
We may need to create a object library with the type sources. Otherwise, we will have two different targets depending on generated files, that will cause problems if we try to use parallel builds

Reply:
Yes! I solved the issue with my last commit.

Thanks!